File: go/0108E-将有序数组转换为二叉搜索树.go
/*
 * @Author: SukiEva
 * @Date: 2021-11-12 11:32:06
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sortedArrayToBST(nums []int) *TreeNode {
	if len(nums) == 0 {
		return nil
	}
	rootIndex := len(nums) / 2
	root := &TreeNode{
		Val: nums[rootIndex],
	}
	if len(nums) == 1 {
		return root
	}
	root.Left = sortedArrayToBST(nums[:rootIndex])
	root.Right = sortedArrayToBST(nums[rootIndex+1:])
	return root
}

File: cpp/0108E-将有序数组转换为二叉搜索树.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-23 11:48:14
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>
#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        int len = nums.size();
        if (len == 1) return new TreeNode(nums[0]);
        int middle = len >> 1;
        TreeNode* T = new TreeNode(nums[middle]);
        if (len == 2)
            T->left = new TreeNode(nums[0]);
        else {
            vector<int> left(nums.begin(), nums.begin() + middle);
            vector<int> right(nums.begin() + middle + 1, nums.end());
            T->left = sortedArrayToBST(left);
            T->right = sortedArrayToBST(right);
        }
        return T;
    }
};